src/VisualStudio/Core/Next/Remote/JsonRpcClient.cs

```diff
@@ -55,7 +53,10 @@ internal class JsonRpcClient : IDisposable
                 // cancellation exception here. if any exception is thrown unrelated to cancellation, then we will rethrow
                 // the exception
                 _cancellationToken.ThrowIfCancellationRequested();
-                throw;
+
+                // this is to make us not crash. we should remove this once we figure out
+                // what is causing this
```

Comment:
I am +1 on non crash (NFW) over crashing. but unfortunately, that is not the way Roslyn is decided on unexpected exception. we can bring up that argument again but this won't change current design on crashing on any unexpected exception rule our team has.

this is very specific case we added temporarily. our default will be still crash once we figure out this issue.